### ISSUE TITLE
read_json with lines=True raises exception on multiple consecutive newlines

### ISSUE THREAD

**Issue by louispotok** [pandas-dev/pandas#17615]
I found this while working on [this pull request](https://github.com/pandas-dev/pandas/pull/17168). I intend to fix it in the same PR, but for thoroughness wanted to open it as its own issue in case there's debate about whether the current behavior is more desirable.

#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
j = """
{"A":1,"B":4}

{"A":2,"B":5}
"""
pd.read_json(j, lines=True)
```
#### Problem description
This raises `ValueError: Expected object or value` in a call to `json.loads` on line 639 of `pandas/io/json/json.py`. This happens in 0.20.2 and on the latest dev version.

#### Expected Output
I would expect this to return a DataFrame as though there were no blank lines in the input.

**Comment by gfyoung** [comment]
@louispotok : Is there a version where this did not raise an error?  Otherwise, it would seem malformed IMO, but then again, maybe this is prevalent enough that we should allow it.

**Comment by louispotok** [comment]
I've only tried it on 20.2 and the latest dev version, so I don't know if it ever worked. I've never encountered it in the wild, just while writing tests for my PR. It's easy for me to imagine situations where it could happen (e.g. by appending a file that starts with a newline to a file that ends with a newline) but I googled a bit and don't see any SO questions or similar, so maybe it isn't common.
